src/engine/bpmcontrol.cpp

```diff
@@ -874,14 +862,9 @@ void BpmControl::collectFeatures(GroupFeatureState* pGroupFeatures) const {
     if (getBeatContextNoLookup(dThisPosition,
                        dThisPrevBeat, dThisNextBeat,
                        &dThisBeatLength, &dThisBeatFraction)) {
-        pGroupFeatures->has_prev_beat = true;
-        pGroupFeatures->prev_beat = dThisPrevBeat;
-
-        pGroupFeatures->has_next_beat = true;
-        pGroupFeatures->next_beat = dThisNextBeat;
-
-        pGroupFeatures->has_beat_length = true;
-        pGroupFeatures->beat_length = dThisBeatLength;
+        pGroupFeatures->has_beat_length_sec = true;
+        // Note: dThisBeatLength is fractional frames count * 2 (stereo samples)  
+        pGroupFeatures->beat_length_sec = dThisBeatLength / m_pTrack->getSampleRate() / 2; 
```

Comment:
I am doubting that this division by 2 should be here. It requires every effect to do this inverse operation (multiply by 2).

Reply:
On second thought it may be better to leave it as is and move effect code towards working with frames.

Reply:
Yes, we should move to frames whenever no external interface change is required.
Playpositions are stored in stereo samples in our database, that can not be changed.